### ISSUE TITLE
Different texture scaling, ceiling at edge of temple interior

### ISSUE THREAD

**Issue by Allofich** [afritz1/OpenTESArena#129]
While testing the latest transition code, I found a spot in one of the temple interiors that looks different in original Arena and OpenTESArena. There is a short wall here at the edge of the map. Here is how it looks in OpenTESArena.

![opentesarena](https://user-images.githubusercontent.com/19624336/42449053-394fb51c-83ba-11e8-96ee-859c254a07ae.png)

And here is how it looks in original Arena.

![arena](https://user-images.githubusercontent.com/19624336/42449260-e8dcc722-83ba-11e8-91c0-f03613523d82.png)

The texture is cut off in OpenTESArena, the size is scaled to the wall in original Arena. Also the blue ceiling ends here and there is a black void in OpenTESArena, while the ceiling continues to stretch on in original Arena.

Location in the temple:
![arena2](https://user-images.githubusercontent.com/19624336/42449445-b1327d3e-83bb-11e8-877e-efd81991636e.png)

You can find one of these temple types at "Brotherhood of Faith" in the Imperial City. It's near the southwest corner.
![a_009](https://user-images.githubusercontent.com/19624336/42449476-c8da72de-83bb-11e8-8ba1-e88409d0ac26.png)

**Comment by afritz1** [comment]
The difference in look for those two textures is because the texture coordinates for raised platforms aren't properly being used yet (`box3a`, `box3b`, and `box4` in [ExeData.h](https://github.com/afritz1/OpenTESArena/blob/bcf9b5cbbd64d6a7d925172c1ba7eda352a9471c/OpenTESArena/src/Assets/ExeData.h#L448)). Those values should be used in the level construction code in [LevelData.cpp](https://github.com/afritz1/OpenTESArena/blob/3463aa307e4a7b88c7aa1ea607ae0c2602f0e368/OpenTESArena/src/World/LevelData.cpp#L501).

I'm on the fence about replicating the out-of-bounds map behavior because it appears that Arena uses some data that's not defined in the .MIF file, perhaps a byproduct of the game having a chunk-based system for level presentation (64x64, etc.). I don't know how to define the out-of-bounds part because it's not obvious what it should be (some default value?). The level code and software renderer would need some redesigning to compensate for the levels not being a static set of voxels.

**Comment by Carmina16** [comment]
Ceiling can be infinite in Arena, as it is just a texture drawn in the top of the screen.

**Comment by afritz1** [comment]
Yeah, the ceilings are easy like Carmina16 said. I'm assuming everything in `MAP1` is just air, too? Not sure about default values for the floor.

So like I said above, it's going to involve some redesigning of the level data to get the "infinite" behavior of interiors in the original game. I still don't like it conceptually (why would an interior have infinite floor/ceiling?), but if the original game does something as obvious as this then OpenTESArena should do it too. Moving over to a chunk-based system will take some time but I think it's doable. Probably won't get to it soon because of higher priority things, like those for 0.8.0.

**Comment by afritz1** [comment]
Almost got the rendering to work in PR #200. Currently dealing with some ray casting math errors but the floor and ceiling of adjacent chunks should be working soon. I'm testing with TEMPLE8.MIF. Any idea what texture IDs they should be? I tried voxel IDs of 2 for floor and 14 for ceiling (only relevant to my engine I guess, since 0 is air) but some interiors don't seem to have that many textures, so it doesn't work. Hoping that they are specified in the .MIF header or something.

**Comment by afritz1** [comment]
This image is after sidestepping through that half wall with collision turned off while facing north. The bending is due to the ray casting error I mentioned. Still working out some other chunk-related math issues too.
![screenshot000](https://user-images.githubusercontent.com/6729557/109602543-260de680-7ad5-11eb-8396-fbc667f9ce7a.png)

**Comment by afritz1** [comment]
Oh right, the ceiling ID is just whatever *CEILING is in the .INF file.

**Comment by afritz1** [comment]
Okay the ceiling rendering is working like it should now.
![screenshot000](https://user-images.githubusercontent.com/6729557/109752139-b5c69a00-7b94-11eb-8738-c50de832e9e0.png)
Here is what it looks like from outside the level:
![screenshot001](https://user-images.githubusercontent.com/6729557/109752165-c0812f00-7b94-11eb-9672-62af1e399dbf.png)
Texture coordinates is a separate issue I'm not looking at right now. Trying to get all the chunk system stuff done. (Also, the entities are temporarily disabled, that's why the brazier isn't there)

**Comment by afritz1** [comment]
The chunk system part of this issue is done (#200), so renaming this issue to be just about texture coordinates.

**Comment by afritz1** [comment]
Still seeing this texture scaling issue on my rendering branch.

**Comment by Carmina16** [comment]
I have some ideas to test about the texture placement.

**Comment by afritz1** [comment]
Okay. This issue is related: https://github.com/afritz1/OpenTESArena/issues/101
